src/services/coronaTrackerService.js

```diff
@@ -0,0 +1,40 @@
+import { EventTargetShim } from '../utils/EventTargetShim.js';
+
+const cache = {};
+
+/**
+ * if force refresh is true
+ * if the previous request failed
+ * if not already in cache
+ */
+
+class CoronaTrackerService extends EventTargetShim {
+  async getCountry(opts) {
+    const { iso2 } = opts;
+    if (opts.forceRefresh || cache[iso2]?.status === 'failed' || !cache[iso2]) {
+      try {
+        cache[iso2] = {};
+        const res = await (await fetch(`https://api.coronatracker.com/v3/stats/worldometer/country?countryCode=${iso2}`)).json();
```

Comment:
This parameter can come from user input, so we should validate if it only contains alphabetical characters and use encodeURIComponent on it. Otherwise someone could send a malicious link.

Reply:
Good catch!